Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.py

```diff
@@ -590,38 +629,75 @@ def format_to_iso(date_string):
     return date_string
 
 
-def fetch_incidents(proxies):
-    last_run = demisto.getLastRun()
-    last_fetch = last_run.get('time')
-
-    # handle first time fetch
-    if last_fetch is None:
-        last_fetch, _ = parse_date_range(date_range=FETCH_TIME, date_format='%Y-%m-%dT%H:%M:%S.%f', utc=False,
-                                         to_timestamp=False)
-        last_fetch = parse(str(last_fetch))
-        last_fetch_timestamp = int(last_fetch.timestamp() * 1000)
+def get_time_range(last_fetch: Union[str, None] = None, fetch_time=FETCH_TIME, time_range_start=None,
+                   time_range_end=None, time_field=TIME_FIELD) -> Dict:
+    """
+    Creates the time range filter's dictionary based on the last fetch and given params.
+    The filter is using timestamps with the following logic:
+        start date (gt) - if this is the first fetch: use time_range_start param if provided, else use fetch time param.
+                          if this is not the fetch: use the last fetch provided
+        end date (lt) - use the given time range end param.
+    Args:
 
-        # if timestamp: get the last fetch to the correct format of timestamp
-        if 'Timestamp' in TIME_METHOD:
-            last_fetch = get_timestamp_first_fetch(last_fetch)
-            last_fetch_timestamp = last_fetch
+        last_fetch (str): last fetch time stamp
+        fetch_time (str): first fetch time
+        time_range_start (str): start of time range
+        time_range_end (str): end of time range
+        time_field ():
 
-    # if method is simple date - convert the date string to datetime
-    elif 'Simple-Date' == TIME_METHOD:
-        last_fetch = parse(str(last_fetch))
-        last_fetch_timestamp = int(last_fetch.timestamp() * 1000)
+    Returns:
+        dictionary (Ex. {"range":{'gt': 1000 'lt':1001}})
+    """
+    range_dict = {}
+    if not last_fetch:  # this is the first fetch
+        # if time range start provided we will use it instead of fetch time.
+        if time_range_start:
+            start_date = dateparser.parse(time_range_start)
+        elif fetch_time:
+            start_date = dateparser.parse(fetch_time)
+        else:
+            raise DemistoException("Missing First fetch timestamp or Time Range Start, please provide one of them.")
```

Comment:
from going over this function code I don't see why we need these two arguments, we can just do the following:
```python
if not last_fetch: # this is the first fetch
    start_date = dateparser.parse(time_range_start)
```
In the `fetch_incidents` you will do:
```python
get_time_range(time_range_start=last_fetch)
```
And `search_command` you will do:
```python
get_time_range(time_range_start=time_range_start)
```